### PR TITLE
meme cards component added

### PR DESCRIPTION
**Pull request by xiningchen**
Cards can be created and a flex box was placed inside browser so that cards are flex items. Inside Browser.html you can see how current cards are created. 
There are still a few things to fix for later:
1) Centring image and clipping
2) Up and down vote needs to be implemented 
3) Take image as an input

### CODE REVIEW

**Comment by ClaytonVis** [comment]
Other than the template.viewer component reference. I think it looks good. Maybe change the background colour of the browse-component (I think). Just so it's a little clearer as to where the cards lay.

File: src/app/meme-card/meme-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  // if cardHeight is > 0, then change CSS
+
+  // if cardWidth is > 0, then change CSS
+
+  voteCount = 0;
+  voted = false;
+
+  // default style
+  cardStyle = {
```
Comment (Step7750) [inline src/app/meme-card/meme-card.component.ts]: In the future if you want to declare like this and overwrite later without worries of type errors, you can do `cardStyle: {};` or `cardStyle: any` (but any is not the best idea...)

File: src/app/meme-card/meme-card.component.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+//  margin: 5px;
+//}
+
+
+.image-frame {
+  overflow: hidden;
+
+  min-width: inherit;
+  max-width: inherit;
+  min-height: 320px;
```
Comment (Step7750) [inline src/app/meme-card/meme-card.component.scss]: `min-height` and `max-height` shouldn't be needed if you set the height explicitly as `height: 320px`. CSS can be funny sometimes though...

File: src/app/meme-card/meme-card.component.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+
+.cardActionButtons {
+  float: right;
+}
+
+.cardActionButtons button {
+  border: 0px;
+  cursor: pointer;
```
Comment (Step7750) [inline src/app/meme-card/meme-card.component.scss]: This isn't needed once `MatButtonModule` is imported.

File: src/app/community/browse/browse.component.html [hunk 4]
```diff
@@ -1,5 +1,13 @@
 <mat-tab-group>
-  <mat-tab label="Top">Content 1</mat-tab>
+  <mat-tab label="Top">
+    <div class="browseContentWrapper">
+      <app-meme-card class="flexItem" *ngFor="let u of users"
+                     [username]=u.un
+                     [cardWidth]=u.width
```
Comment (Step7750) [inline src/app/community/browse/browse.component.html]: Why do we need to pass the width?

Comment (xiningchen) [reply src/app/community/browse/browse.component.html]: uh, because idk how you wanted the width to be calculated. lol. But does your other PR fix this?

Comment (Step7750) [reply src/app/community/browse/browse.component.html]: Yes

**Comment by Niustein** [comment]
Looking good

**Pull request merged**